### ISSUE TITLE
Regarding NPMI Score

### ISSUE THREAD

**Issue by raymondzmc** [MilaNLProc/contextualized-topic-models#120]
Just a general question about the NPMI score used in your implementation.

In the original ProdLDA paper, their reported NPMI score is much higher than score in your paper:
https://arxiv.org/pdf/1703.01488.pdf (Table 1)

I'm wondering if there are any differences in the way NPMI is calculated?

**Comment by vinid** [comment]
Hello!

in the ProdLDA paper, they do not average over the 5 different topic numbers [25, 50, ...], their results are for 50 and 200 topics.

You might also want to look at 1) the preprocessing of the dataset and 2) the implementation (https://github.com/estebandito22/PyTorchAVITM). The other parameters should be the same.

Hope this helps!

**Comment by raymondzmc** [comment]
Hi Federico, thanks for your quick reply. I'm wondering if there are some differences in parameters, because in your [ACL paper](https://aclanthology.org/2021.acl-short.96.pdf), you reported an NPMI of 0.11 for 50 topics on the 20News dataset, while the ProdLDA paper reported 0.24.

There are also similar papers that reported NPMI on the same scale as the ProdLDA papers:
- https://aclanthology.org/D18-1096.pdf
- https://aclanthology.org/P19-1640.pdf
- https://aclanthology.org/2021.acl-long.299.pdf

I believe there are some differences in settings (e.g. window size, normalization), if I find out, I will let you know!

**Comment by vinid** [comment]
Not sure if it makes a difference but in the ProdLDA paper they use only the 20NG training data (11K docs), while we use the entire thing (18K docs).

**Comment by raymondzmc** [comment]
I see, that makes a lot of sense, I assume the vocab would also be different.
